test/e2e_node/restart_test.go

```diff
@@ -474,6 +474,73 @@ var _ = SIGDescribe("Restart", framework.WithSerial(), framework.WithSlow(), fra
 				return checkMirrorPodDisappear(ctx, f.ClientSet, pod.Name, pod.Namespace)
 			}, f.Timeouts.PodDelete, f.Timeouts.Poll).Should(gomega.BeNil())
 		})
+		// Regression test for an extended scenario for https://issues.k8s.io/123980
+		ginkgo.It("should kill running pods that were satisfied the affinity before kubelet restart", func(ctx context.Context) {
```

Comment:
nit: `should evict running pods that do not meet the affinity after the kubelet restart`

Reply:
done